src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs

```diff
@@ -348,6 +648,101 @@ private static double[] ComputeSequenceDeltas(IReadOnlyList<double> a, IReadOnly
             return delta;
         }
 
+        private static ImmutableDictionary<string, TResult>
+            PermutationFeatureImportance<TMetric, TResult>(
+                IHostEnvironment env,
+                ITransformer model,
+                IDataView data,
+                Func<TResult> resultInitializerParam,
+                Func<TMetric, TMetric, TMetric> deltaFuncParam,
+                Func<IDataView, TMetric> evaluationFuncParam,
+                bool useFeatureWeightFilter,
+                int? numberOfExamplesToUse,
+                int permutationCount) where TResult : IMetricsStatistics<TMetric>
+        {
+            env.CheckValue(data, nameof(data));
+            env.CheckValue(model, nameof(model));
+
+            ITransformer lastTransformer = null;
+
+            if (model is TransformerChain<ITransformer> chain)
+            {
+                foreach (var transformer in chain.Reverse())
+                {
+                    if (transformer is ISingleFeaturePredictionTransformer)
+                    {
+                        lastTransformer = transformer;
+                        break;
+                    }
+                }
+            }
+            else lastTransformer = model as ISingleFeaturePredictionTransformer;
+
+            env.CheckValue(lastTransformer, nameof(lastTransformer));
+
+            var lastTransformerType = lastTransformer.GetType();
+            string featureColumnName = (lastTransformer as ISingleFeaturePredictionTransformer).FeatureColumnName;
+            Type pfiType = typeof(PermutationFeatureImportance<,,>);
+
+            TryGetImplementedIPredictionTransformer(lastTransformerType, out lastTransformerType);
```

Comment:
Don't reuse the `lastTransformerType` variable here. It is confusing that the variable change from `lastTransformer.GetType()` to an interface implemented on that type. Instead create a new variable, possibly named `predictionTransformerGenericType`.

Reply:
I'll make that change.